src/commands/play.rs

```diff
@@ -102,6 +118,42 @@ pub async fn execute_play(
     Ok(())
 }
 
+async fn calculate_time_until_play(queue: &[TrackHandle], flag: &PlayFlag) -> Option<Duration> {
```

Comment:
Very nice function, will become even better when we add the error propagation mechanisms and just unwrap values with `?`

Reply:
100% 🚀